### PR TITLE
Powershell module_utils loader and tests

### PR DESCRIPTION
**Pull request by nitzmahone**
##### SUMMARY
* supports custom module_utils loads (anything in module prefaced with `#Requires -Module Ansible.ModuleUtils.*`)
* supports all usual PluginLoader locations (built-in lib/ansible/module_utils/powershell/*, custom path from config, playbook module_utils/powershell, ~/.ansible/module_utils/powershell, role module_utils/powershell, etc),

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
module_common, PluginLoader

##### ANSIBLE VERSION
2.4.0

### CODE REVIEW

File: test/integration/targets/win_module_utils/library/uses_local_utils.ps1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#! powershell
+
+# use different cases, spacing and plural of 'module' to exercise flexible powershell dialect
+#ReQuiReS -Module Ansible.ModuleUtils.PowerShellLegacy
+#Requires -Module Ansible.ModuleUtils.ValidTestModule
```
Comment (jborean93) [inline test/integration/targets/win_module_utils/library/uses_local_utils.ps1]: Did you mean to try a different form here, the comment seems to indicate you do

File: lib/ansible/executor/module_common.py [hunk 2]
```diff
@@ -786,20 +786,25 @@ def _find_module_utils(module_name, b_module_data, module_path, module_args, tas
         lines = b_module_data.split(b'\n')
         module_names = set()
 
-        requires_module_list = re.compile(r'(?i)^#requires \-module(?:s?) (.+)')
+        requires_module_list = re.compile(r'(?i)^#(?:s*)requires \-module(?:s*) (Ansible\.ModuleUtils\..+)')
```
Comment (jborean93) [inline lib/ansible/executor/module_common.py]: Why do we need the `(?:s*)` part in this regex in the beginning? Do we want to support `-module` and `-modules`?

**Review by jborean93 — COMMENTED** [review]
Just a few questions otherwise seems cool to me.

File: test/integration/targets/win_module_utils/library/uses_local_utils.ps1 (hunk 1 above)
Comment (nitzmahone) [reply test/integration/targets/win_module_utils/library/uses_local_utils.ps1]: Yeah, I had just removed the actual different form temporarily and forgot to fix it. Fixed

File: lib/ansible/executor/module_common.py (hunk 2 above)
Comment (nitzmahone) [reply lib/ansible/executor/module_common.py]: Late-afternoon regex glitch, fixed to support the proper forms

**Pull request merged**